diff --git a/gfx/webrender_bindings/RenderCompositorEGL.cpp b/gfx/webrender_bindings/RenderCompositorEGL.cpp
--- a/gfx/webrender_bindings/RenderCompositorEGL.cpp
+++ b/gfx/webrender_bindings/RenderCompositorEGL.cpp
@@ -154,4 +154,6 @@
   if (mEGLSurface != EGL_NO_SURFACE && aDirtyRects.Length() > 0) {
-    gfx::IntRegion bufferInvalid;
+    const auto fullDamageSize = GetBufferSize();
+    gfx::IntRegion bufferInvalid(gfx::IntRect(
+        0, 0, fullDamageSize.width, fullDamageSize.height));
     const auto bufferSize = GetBufferSize();
     for (const DeviceIntRect& rect : aDirtyRects) {